presto-main/src/main/java/io/prestosql/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -341,24 +342,29 @@ protected Node visitShowSchemas(ShowSchemas node, Void context)
         @Override
         protected Node visitShowCatalogs(ShowCatalogs node, Void context)
         {
-            List<Expression> rows = listCatalogs(session, metadata, accessControl).keySet().stream()
+            ImmutableList.Builder<Expression> rows = ImmutableList.builder();
```

Comment:
This can be structure as follows. It's simpler and clearer what's going on:

```java
List<Expression> rows = listCatalogs(session, metadata, accessControl).keySet().stream();

Optional<Expression> predicate = Optional.empty();
if (rows.isEmpty()) {
    rows = ImmutableList.of("");
    predicate = Optional.of(BooleanLiteral.FALSE_LITERAL);
}
else if (node.getLikePattern().isPresent()) {
    predicate = Optional.of(new LikePredicate(...));
}

return simpleQuery(...)
```